modules/imgproc/src/intersection.cpp

```diff
@@ -51,15 +51,15 @@ static int _rotatedRectangleIntersection( const RotatedRect& rect1, const Rotate
 {
     CV_INSTRUMENT_REGION();
 
-    // L2 metric
-    const float samePointEps = std::max(1e-16f, 1e-6f * (float)std::max(rect1.size.area(), rect2.size.area()));
-
     Point2f vec1[4], vec2[4];
     Point2f pts1[4], pts2[4];
 
     rect1.points(pts1);
     rect2.points(pts2);
 
+    // L2 metric
+    float samePointEps = 1e-6f * (float)std::max(rect1.size.area(), rect2.size.area());
```

Comment:
We don't really want to have zero or denormalized zero value here.

Reply:
What do you mean ? this is not original code, I just moved it.